test/webdriverio/test/basic_test.ts

```diff
@@ -363,4 +363,26 @@ suite('Keyboard navigation on Fields', function () {
     // The same field should still be focused
     chai.assert.equal(await getFocusedFieldName(this.browser), 'BOOL');
   });
+
+  test('Do not reopen field editor when handling enter to make a choice inside the editor', async function () {
+    // Open colour picker
+    await focusOnBlockField(this.browser, 'colour_picker_1', 'COLOUR');
+    await this.browser.pause(PAUSE_TIME);
+    await this.browser.keys(Key.Enter);
+    await this.browser.pause(PAUSE_TIME);
+
+    // Move right to pick a new colour.
+    await keyRight(this.browser);
+    // Enter to choose.
+    await this.browser.keys(Key.Enter);
+
+    // Focus seems to take longer than a single pause to settle.
+    await this.browser.waitUntil(
+      () =>
+        this.browser.execute(() =>
+          document.activeElement?.classList.contains('blocklyActiveFocus'),
```

Comment:
```suggestion
          document.activeElement?.classList?.contains('blocklyActiveFocus'),
```

Is this needed to avoid a potential null?

Reply:
I don't think so, e.g. testing in Node:

```node
> document = { activeElement: null }
{ activeElement: null }
> document.activeElement?.classList.contains('foo')
undefined
```

Reply:
To add to matt's answer, if the activeElement exists, the classList will exist too (it might be an empty list, but it won't be null) https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

Reply:
Ah. This behaves differently in Kotlin land (where I'm much more familiar). There the '?' actually cascades. In this case, `document.activeElement?.classList` would evaluate to either `null` or the class list which is why the additional `?` would be needed (since it would be trying to call `contains` on `null` rather than avoiding the call altogether).

This will take me some time to get used to. :)

Reply:
Yeah, the way optional chaining works in JS is very pragmatic from the point of view of programmers but decidedly weird from the point of view of JS implementers.  I found the parse trees that result from expressions with optional chaining to be _extremely_ unintuitive—despite being (or perhaps because I was) already very familiar with ES5 parsing.